### PR TITLE
VTAdmin: Support for schema migrations view/create

### PR DESCRIPTION
**Pull request by beingnoble03**
## Description
This PR adds a screen for schema migrations list for a particular keyspace, and a screen to create schema migration request. Also includes some related API changes.

## Screenshots
Screenshot 2024-11-04 at 7 56 55 PM
Screenshot 2024-11-08 at 3 25 40 PM

## Related Issue(s)
- #11690

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Hey there! A few comments:

- DDL strategy can include DDL strategy flags. So for example it could look like `vitess --postpone-completion --singleton-context --cut-over-threshold=15s`. In light of that, let leave it editable. But then, it also requires quite a length box. I'd say `keyspace` has too much space, and DDL strategy too little.
- It is possible to submit a migration with `direct` DDL strategy -- but then it won't show in the schema migrations page.
- However, the strategy `mysql` is a hybrid of the two - and will show in the schema migrations page.

So yes, let's allow free text in the DDL strategy flag.

**Comment by beingnoble03** [comment]
As discussed, I've removed the `ddl-strategy` text box, to allow only `vitess` ddl strategy. We can add the advanced options in a follow-up PR?

**Comment by rohit-nayak-ps** [comment]
> > As discussed, I've removed the ddl-strategy text box, to allow only vitess ddl strategy.
> 
> This looks to be a misunderstanding, in [#17134 (comment)](https://github.com/vitessio/vitess/pull/17134#issuecomment-2457134009) I indicated that we should keep it after all?

My bad. The plan was to create an initial PR with defaults and then add UI for the strategy options. But maybe we should just add the text box to start with, so that we do support everything and then replace them with the UI later. @beingnoble03, can you please reinstate the text box, sorry for the confusion.

**Comment by shlomi-noach** [comment]
@rohit-nayak-ps no problem, feel free to lead the way. I'll set some time next week to think about what we discussed over zoom, and how the UI can take shape in the next iteration.

**Comment by beingnoble03** [comment]
@rohit-nayak-ps @shlomi-noach done, also increased the size of the box as mentioned.

**Comment by rohit-nayak-ps** [comment]
* On creating a new migration, it goes to the Migrations screen but without choosing a keyspace, so it shows up empty and is confusing. Also on refreshing it goes back to the empty screen because it doesn't remember the chosen keyspace.

* We should also have a details page for each migration with more details including the actual sql query.

* i created a migration in a sharded keyspace and it only shows one of the shards in the detail.

* Also I think we should change the default strategy to `vitess`: since that is how we expect most users to run migrations. Remove gh-ost (already unsupported) and pt-osc (slated to be removed soon): https://vitess.io/docs/22.0/user-guides/schema-changes/ddl-strategies/.

* Regarding UI: can we align the timestamps so that it is easier to compare.

**Review by rohit-nayak-ps — APPROVED** [review]
Nice. Let's should do the following in a followup:
* Show a single line for each migration, keyed with UUID. That line can show just the shards involved and a consolidated status (similar to how we show a combined Workflow status for all streams)
* Add a details page where all the shards involved along with more info like the actual query itself is shown

File: web/vtadmin/src/util/schemaMigrations.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { invertBy } from 'lodash-es';
+import { vtctldata } from '../proto/vtadmin';
+
+/**
+ * SCEMA_MIGRATION_STATUS maps numeric schema migration status back to human readable strings.
```
Comment (notfelineit) [inline web/vtadmin/src/util/schemaMigrations.ts:20]:
```suggestion
 * SCHEMA_MIGRATION_STATUS maps numeric schema migration status back to human readable strings.
```

File: web/vtadmin/src/util/schemaMigrations.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export const SCEMA_MIGRATION_STATUS = Object.entries(invertBy(vtctldata.SchemaMigration.Status)).reduce(
+    (acc, [k, vs]) => {
+        acc[k] = vs[0];
+        return acc;
+    },
+    {} as { [k: string]: string }
+);
+
+export const formatSchemaMigrationStatus = (schemaMigration: vtctldata.ISchemaMigration) =>
+    schemaMigration.status && SCEMA_MIGRATION_STATUS[schemaMigration.status];
```
Comment (notfelineit) [inline web/vtadmin/src/util/schemaMigrations.ts:31]:
```suggestion
    schemaMigration.status && SCHEMA_MIGRATION_STATUS[schemaMigration.status];
```

**Review by notfelineit — APPROVED** [review]

File: web/vtadmin/src/components/routes/createSchemaMigration/CreateSchemaMigration.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+const CALLER_ID_HELP_TEXT =
+    'Effective caller ID used for the operation and should map to an ACL name which grants this identity the necessary permissions to perform the operation (this is only necessary when strict table ACLs are used)';
+
+export const CreateSchemaMigration = () => {
+    useDocumentTitle('Create Schema Migration Request');
+
+    const history = useHistory();
+
+    const [formData, setFormData] = useState<FormData>(DEFAULT_FORM_DATA);
```
[the remark this answers is not in the document]
Comment (beingnoble03) [reply web/vtadmin/src/components/routes/createSchemaMigration/CreateSchemaMigration.tsx:70]: 👍  I think this makes more sense to be done in a separate PR, as there are other components using the same structure.

**Pull request merged**